File: sgu/486.cpp
#include<bits/stdc++.h>
using namespace std;
int vis[10];
char s1[10],s2[10];
int main(){
    scanf("%s%s",s1,s2);
    for(int i=0;i<4;i++){
        vis[s1[i]-'0']=1;
    }
    int ans1=0,ans2=0;
    for(int i=0;i<4;i++){
        if(s1[i]==s2[i])ans1++;
        else if(vis[s2[i]-'0'])ans2++;
    }
    printf("%d %d",ans1,ans2);
    return 0;
}

File: sgu/486.java
import java.io.*;
import java.math.BigInteger;
import java.util.Scanner;

public class Main{
    public static void main(String[] args)throws Exception{
        Scanner in = new Scanner(System.in);
        int c[]=new int[10];
        for(int i=0;i<10;i++)c[i]=0;
        String s1=in.next();
        String s2=in.next();
        for(int i=0;i<4;i++){
            c[s1.charAt(i)-'0']=1;
        }
        int ans1=0;
        int ans2=0;
        for(int i=0;i<4;i++){
            if(s1.charAt(i)==s2.charAt(i))ans1++;
            else if(c[s2.charAt(i)-'0']==1)ans2++;
        }
        System.out.println(ans1+" "+ans2);
    }
}
